src/harps-n/util.jl

```diff
@@ -0,0 +1,172 @@
+    df = DataFrame(order=Int[],pixels=UnitRange[])
+    for ord in orders
+        pixel_array = EchelleInstruments.calc_complement_index_ranges(get_pixel_range(NEID2D(),ord),EchelleInstruments.NEID.bad_col_ranges(NEID2D(),ord))
+        order_array = repeat([ord],length(pixel_array))
+        df_tmp = DataFrame(:order=>order_array,:pixels=>pixel_array)
+        append!(df,df_tmp)
+    end
+    return df
+end
+=#
+
+#=
+function make_clean_line_list_from_bad_columns(line_list::DataFrame, neid_data::DT ;
+                Δv_to_avoid_tellurics::Real = 0.0, v_center_to_avoid_tellurics::Real = 0.0
+                )  where { T1<:Real, A1<:AbstractArray{T1}, T2<:Real, A2<:AbstractArray{T2}, T3<:Real, A3<:AbstractArray{T3}, IT<:NEID.AnyNEID, ST<:Spectra2DBasic{T1,T2,T3,A1,A2,A3,IT}, DT<:AbstractArray{ST,1} }
+
+                if size(df_bad_col_λs,1) < 1 return line_list end
+                line_list_to_search_for_tellurics = copy(line_list)
+                line_list_to_search_for_tellurics.lambda_lo = line_list_to_search_for_tellurics.lambda./calc_doppler_factor(Δv_to_avoid_tellurics).*calc_doppler_factor(v_center_to_avoid_tellurics)
+                line_list_to_search_for_tellurics.lambda_hi = line_list_to_search_for_tellurics.lambda.*calc_doppler_factor(Δv_to_avoid_tellurics).*calc_doppler_factor(v_center_to_avoid_tellurics)
+
+                line_list_no_tellurics_df = line_list_to_search_for_tellurics |> # @filter(λmin <= _.lambda <= λmax) |>
+                   @filter( !RvSpectMLBase.is_in_wavelength_range_list(_.lambda, order=_.order, list=df_bad_col_λs ) ) |>
+                   DataFrame
+end
+=#
+
+
+#old neid functions
+
+
+"""
+   `make_clean_line_list_from_tellurics(line_list, harpsn_data; Δv_to_avoid_tellurics )`
+Returns a new line list that excludes lines with telluric contamination.
+Inputs:
+- `line_list`:  Dataframe containing field lambda
+# - `harpsn_data`:  Array of spectra
+- `Δv_to_avoid_tellurics`:  in m/s
+- `v_center_to_avoid_tellurics` : in m/s
+- `tellurics_filename`:  filename with wavelength ranges affected by tellurics
+Outputs:
+- `line_list_without_tellurics`:   DataFrame with fields: `lambda`, `weight`, `lambda_lo`, and `lambda_hi`.
+Warning: Currently, reads in a list of wavelength ranges with tellurics and this file was created based on tellurics metadata in EXPRES data for HR 101501.
+"""
+function make_clean_line_list_from_tellurics(line_list::DataFrame, harpsn_data::DT ;
+    Δv_to_avoid_tellurics::Real = 0.0, v_center_to_avoid_tellurics::Real = 0.0, tellurics_filename::String = joinpath(pkgdir(EchelleInstruments),"data/neid/telluric_ranges.csv")
+               ) where { T1<:Real, A1<:AbstractArray{T1}, T2<:Real, A2<:AbstractArray{T2}, T3<:Real, A3<:AbstractArray{T3}, IT<:HARPSN.AnyHARPSN, ST<:Spectra2DBasic{T1,T2,T3,A1,A2,A3,IT}, DT<:AbstractArray{ST,1} }
+   @assert 0.5*RvSpectMLBase.max_bc_earth_rotation <= Δv_to_avoid_tellurics <= 4*RvSpectMLBase.max_bc
+
```

Comment:
Let's add an @error "Some message that tells people we know this shouldn't work yet" here until it's working for HARPSN